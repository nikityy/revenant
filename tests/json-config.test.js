const fs = require("fs");
const path = require("path");
const { promisify } = require("util");

const removeFile = promisify(fs.unlink);

const JSONConfig = require("../lib/json-config");

const configPath = path.join(__dirname, "config.json");
const tempConfigPath = path.join(__dirname, "temp-config.json");
const unexistingConfigPath = path.join(__dirname, "unexisting-config.json");

describe("readConfigFile", () => {
  test("should read config from a json file", async () => {
    const config = await new JSONConfig(configPath).readConfig();

    expect(config).toMatchSnapshot();
  });

  test("should fallback to empty config if file does not exist", async () => {
    const config = await new JSONConfig(unexistingConfigPath).readConfig();

    expect(config).toMatchSnapshot();
  });
});

describe("writeConfigFile", () => {
  afterEach(() => removeFile(tempConfigPath));

  test("should correctly serialize and write to file", async () => {
    const originalConfig = await new JSONConfig(configPath).readConfig();

    await new JSONConfig(tempConfigPath).saveConfig(originalConfig);

    const newConfig = await new JSONConfig(tempConfigPath).readConfig();

    expect(newConfig).toEqual(originalConfig);
  });
});
