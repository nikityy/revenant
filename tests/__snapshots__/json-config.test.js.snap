// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readConfigFile should fallback to empty config if file does not exist 1`] = `
Object {
  "downloadPath": null,
  "kinopoisk": Object {
    "watchUrl": null,
  },
  "rutracker": Object {
    "cookie": null,
  },
  "torrents": Object {},
  "version": "1",
  "watchlist": Array [],
}
`;

exports[`readConfigFile should read config from a json file 1`] = `
Object {
  "downloadPath": "~/Downloads/",
  "kinopoisk": Object {
    "watchUrl": "https://www.kinopoisk.ru/user/789114/movies/list/type/3575/#list",
  },
  "rutracker": Object {
    "cookie": "bb-cookie:123",
  },
  "torrents": Object {
    "a": Array [
      "1",
      "2",
      "3",
    ],
    "b": Array [
      "4",
      "5",
    ],
    "c": Array [
      "6",
      "7",
      "1",
    ],
  },
  "version": "1",
  "watchlist": Array [
    "revenant",
    "who's afraid of virginia woolf",
  ],
}
`;
